v8.go

```diff
@@ -163,7 +163,22 @@ func NewIsolate() *Isolate {
 // to initialize all Contexts created from this Isolate.
 func NewIsolateWithSnapshot(s *Snapshot) *Isolate {
 	v8_init_once.Do(func() { C.v8_init() })
-	iso := &Isolate{ptr: C.v8_Isolate_New(s.data), s: s}
+	iso := &Isolate{ptr: C.v8_Isolate_New(s.data, nil), s: s}
+	runtime.SetFinalizer(iso, (*Isolate).release)
+	return iso
+}
+
+type ResourceConstraints struct {
+	// MaxOldSpaceSize sets the maximum size of the old object heap in MiB.
+	MaxOldSpaceSize int
```

Comment:
Please doc the behavior when the resource limits are hit.  That it crashed the process was a surprise!  I'm happy that you're adding the OOM handler.